src/conversion.jl

```diff
@@ -73,5 +72,6 @@ function Base.convert(S::Type{<:Union{Number, T}}, p::APL{T}) where T
     end
     s
 end
+Base.convert(::Type{T}, p::APL) where T<:Number = scalarize(T, p)
```

Comment:
Does this still cause some invalidations ?

Reply:
Nothing serious, if memory serves. The real problem is totally-unconstrained `T`. 

When fixing invalidations, usually I don't like to suggest restrictions on what packages can do (better to fix the code being invalidated). But `convert` is one of those rare functions that's used in a ton of places with poor inference on the 2nd argument, and it's often used to make inference better. Consequently it's bad to specialize only on the 2nd argument, since so many callers don't know what type their 2nd argument is. It's generally fine if you specialize on the first argument.

You can add more `convert` methods than this, I just gave one that is needed to pass tests. But keep in mind that the call site could also just call `scalarize` directly, and that would be the cleanest solution of all (just deleting this method altogether).

Reply:
Ok, thanks for the detailed explanation. I try to keep the same behavior independently of whether the coefficient is a number or not as the coefficient is for instance sometimes a JuMP expressions which is not a number. Therefore I might remove this `convert` and see what's break. Meanwhile I'll merge this PR. Thanks!